frontend/app/services/cardService.ts

```diff
@@ -23,7 +30,10 @@ export const fetchCards = async (
   const url = new URL(API_ROUTES.IDIOMS, IDIOMS_BACKEND_URL);
   url.searchParams.append('page', page.toString());
   url.searchParams.append('limit', limit.toString());
-  if (search) url.searchParams.append('search', search.trim());
+
+  if (search && search.trim().length > 2) {
+    url.searchParams.append('search', search.trim());
+  }
```

Comment:
This is all we should need, please remove any other input validation for the search query in other places